tests/fuzz/libfuzz.h

```diff
@@ -0,0 +1,18 @@
+#ifndef LIGHTNING_TESTS_FUZZ_LIBFUZZ_H
+#define LIGHTNING_TESTS_FUZZ_LIBFUZZ_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+/* Called once before running the target. Use it to setup the testing
```

Comment:
do we also need a teardown function for doing cleanup?

Reply:
Most fuzzing engines offer this API:
- init
- run
I'd rather not differ too much from this pattern